src/lib/tx/transaction.rs

```diff
@@ -0,0 +1,445 @@
+// Copyright (c) DUSK NETWORK. All rights reserved.
+// Licensed under the MPL 2.0 license. See LICENSE file in the project root for details.
+
+//! Phoenix transaction structure implementation.
+
+use crate::tx::{Crossover, Fee};
+use dusk_plonk::bls12_381::Scalar as BlsScalar;
+use dusk_plonk::proof_system::Proof;
+use phoenix_core::note::Note;
+use phoenix_core::Error;
+use std::convert::{TryFrom, TryInto};
+use std::io::{self, Read, Write};
+use std::mem::transmute_copy;
+
+/// Type identifiers for Phoenix transactions.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum TxType {
+    Transfer,
+    Distribute,
+    WithdrawFees,
+    Bid,
+    Stake,
+    Slash,
+    WithdrawStake,
+    WithdrawBid,
+}
```

Comment:
Type has nothing to do with the contract that you're calling. 

Therefore, this should be removed.